lib/QubitXmlImport.class.php

```diff
@@ -658,7 +658,7 @@ private function processMethods(&$domNode, &$importDOM, $methods, &$currentObjec
                         if (null !== ($node = eval('return '.$str.';')))
                         {
                           // Substitute node value for search string
-                          $parameter = str_replace($match, '\''.$node->nodeValue.'\'', $parameter);
+                          $parameter = str_replace($match, '\''.addcslashes($node->nodeValue, "'").'\'', $parameter);
```

Comment:
As far as I can tell, this allows arbitrary execution (by PHP) of *any* string used for `$node->nodeValue` in line 671 below. :(

Reply:
True. I think most of our XML importing relies on this mechanism, which is not so good.

Reply:
@mcantelon will be creating a new ticket describing the security issue with the eval() use.

Reply:
Ticket describing eval() issue:
https://projects.artefactual.com/issues/13216